### DISCUSSION TITLE
Rate limiting / throttling

### DISCUSSION THREAD

**Discussion by jmnunezizu** [unitycatalog/unitycatalog#680] [Ideas]
The Hive metastore (HMS) is subject to "noisy neighbour" problems. In other words, a user of the system could place load on the system's resources that could have an adverse effect on other users of the Data Lake.

The Hive metastore does not support rate limiting. Therefore, we had to implement rate limiting in the federation layer on top of HMS: Waggle Dance. Please see https://github.com/ExpediaGroup/waggle-dance/blob/main/waggle-dance-extensions/README.md.

I went through the proposed roadmap page (https://github.com/unitycatalog/unitycatalog/discussions/411) and did not find any reference/mention. 

Given that UC has REST APIs, it might not be required to have rate limiting built into it. Other solutions, like sidecars, could be used instead. In any case, discussing the problem and deciding the path forward, either via documentation/recommendation or implementing the feature, will be beneficial.

Thanks,
Jose.-

**Comment by dennyglee** [comment]
Thanks @jmnunezizu - I definitely like the idea!  Let's convert this to a discussion so we can dive into this, eh?!

**Reply by jmnunezizu** [reply]
Sounds good! Thanks, @dennyglee.
